specification/security/resource-manager/common/v1/types.json

```diff
@@ -114,6 +114,91 @@
     "AscLocationProperties": {
       "type": "object",
       "description": "An empty set of properties"
+    },
+    "ResourceDetails": {
+      "type": "object",
+      "description": "Details of the resource that was assessed",
+      "discriminator": "source",
+      "properties": {
+        "source": {
+          "type": "string",
+          "description": "The platform where the assessed resource resides",
+          "enum": [
+            "Azure",
+            "OnPremise"
+          ],
+          "x-ms-enum": {
+            "name": "source",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Azure",
+                "description": "Resource is in Azure"
+              },
+              {
+                "value": "OnPremise",
+                "description": "Resource in an on premise machine connected to Azure cloud"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "source"
+      ]
+    },
+    "AzureResourceDetails": {
+      "type": "object",
+      "description": "Details of the Azure resource that was assessed",
+      "x-ms-discriminator-value": "Azure",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceDetails"
+        }
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Azure resource Id of the assessed resource"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "OnPremiseResourceDetails": {
```

Comment:
You aren't using this type, it can be removed.

Reply:
this and the "AzureResourceDetails" type are not referenced here directly, but the assessment object has a reference to their mutual basic abstract type, "ResourceDetails"
and we use the "discriminator" and "x-ms-discriminator-value" to differentiate between the two